agenta-web/src/lib/helpers/analytics/AgPosthogProvider.tsx

```diff
@@ -0,0 +1,51 @@
+import {useCallback, useEffect, useRef} from "react"
+import {useRouter} from "next/router"
+import {useAtom} from "jotai"
+import {posthogAtom, type PostHogConfig} from "./store/atoms"
+import {CustomPosthogProviderType} from "./types"
+
+const CustomPosthogProvider: CustomPosthogProviderType = ({children, config}) => {
+    const router = useRouter()
+    const loadingPosthog = useRef(false)
+    const [posthogClient, setPosthogClient] = useAtom(posthogAtom)
+
+    const initPosthog = useCallback(async () => {
+        if (!!posthogClient) return
+        if (loadingPosthog.current) return
+
+        loadingPosthog.current = true
```

Comment:
Quick note: the `loadingPosthog` ref isn't being reset, which could cause issues if initialization fails